### PR TITLE
V3: Add Graph Tile automation tests

### PR DESCRIPTION
**Pull request by tejal-shah**
@kswenson , @eireland ,
Below are the changes in this PR:
- Added tests for Graph tile (various plot transitions, but not checking against image snapshots yet), tests for Axes and Legend
- Added cypress support for being able to run tests in staging (https://codap3.concord.org/index-staging.html) in addition to dev (https://codap3.concord.org/branch/main/) and local
- Added MouseSensor to `v3/src/components/codap-dnd-context.tsx` - without this I couldn't get drag and drop to work as cited here - https://github.com/clauderic/dnd-kit/issues/208 . But unfortunately, this is breaking some functionality in the case table like renaming an attribute, opening `Edit Attribute Properties...` modal, hiding an attribute, deleting an attribute - this list might not be exhaustive, but these are the ones I'm aware of. I have skipped a few table tests that test these features just to have a deployable version of this problem so I can show the issue and seek your help. And since breaking this functionality is not desirable at all, **we should not merge this PR until we fix this**. But I'll need help from one of you in making this MouseSensor to work without breaking these features since I need it to get the drag and drop cypress tests to work.

You'll see this happening in the deployed version of this branch here - https://codap3.concord.org/branch/graph-automation-tests/?sample=mammals
Screencast - https://www.screencast.com/t/NBxm947C77av

### CODE REVIEW

**Review by kswenson — CHANGES_REQUESTED** [review]
👍 It's great that we're this close to having cypress tests that work for the graph and in particular for drag/drop. We obviously need to solve the issues you raise, but it seems like you're on the right track. I made some suggestions in inline comments but if necessary we can get together as well.

File: v3/cypress/support/commands.ts [hunk 1]
```diff
@@ -1,3 +1,122 @@
...
+            .wait(options?.delay || 0, { log: Boolean(options?.delay) })
+            .trigger("mousemove", {
+                force: true,
+                clientX: Math.floor(targetRect.x + targetRect.width / 2),
+                clientY: Math.floor(targetRect.y + targetRect.height / 2),
+            })
+            .wait(options?.delay || 0, { log: Boolean(options?.delay) })
+            .trigger("mouseup", { force: true })
+            .wait(options?.delay || 0, { log: Boolean(options?.delay) })
+    })
```
[the remark this answers is not in the document]
Comment (kswenson) [reply v3/cypress/support/commands.ts:122]: I did a simple test and the "pointerMoveBy" command didn't work as well as the "mouseMoveBy" command. 😞 So I went ahead and implemented the url parameter for enabling the mouse sensor only for tests that require it.

File: v3/src/components/codap-dnd-context.tsx [hunk 2]
```diff
@@ -12,7 +12,8 @@ export const CodapDndContext = ({ children }: IProps) => {
   const sensors = useSensors(
                     // pointer must move three pixels before starting a drag
                     useSensor(PointerSensor, { activationConstraint: { distance: 3 }}),
-                    useSensor(KeyboardSensor, { coordinateGetter: customCoordinatesGetter }))
+                    useSensor(KeyboardSensor, { coordinateGetter: customCoordinatesGetter }),
+                    useSensor(MouseSensor))
```
[the remark this answers is not in the document]
Comment (kswenson) [reply v3/src/components/codap-dnd-context.tsx]: I went ahead and added the url parameter for enabling the `MouseSensor` on demand for cypress tests.

File: v3/cypress/e2e/attribute_types_test.spec.ts [hunk 3]
```diff
@@ -4,7 +4,8 @@ import {CfmElements as cfm} from "../support/elements/CfmObject"
 context("attribute types", () => {
     before(() => {
         const filename = "cypress/fixtures/attribute-types-test-document.codap"
-        cy.visit("")
+        const url = `${Cypress.config("index")}`;
```
[the remark this answers is not in the document]
Comment (kswenson) [reply v3/cypress/e2e/attribute_types_test.spec.ts]: done

**Review by kswenson — CHANGES_REQUESTED** [review]
I went ahead and added the `mouseSensor` url parameter for enabling the `MouseSensor` on demand. There's a couple of things worth addressing before merging the PR:
- we've standardized on hyphens rather than underscores for filenames, so cypress test filenames should follow that convention
- some of the legend tests were failing when I did a quick test

File: v3/cypress/support/commands.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tejal-shah) [reply v3/cypress/support/commands.ts:122]: Yeah, that was the first thing I had tried before even adding the mousedown, mouseup and mousemove here.

File: v3/src/components/codap-dnd-context.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (tejal-shah) [reply v3/src/components/codap-dnd-context.tsx]: Yay, thank you!

File: v3/package.json [hunk 4]
```diff
@@ -63,6 +63,8 @@
     "test:watch": "jest --watch",
     "test:coverage:watch": "jest --coverage --watchAll",
     "test:cypress": "cypress run",
+    "test:cypress:dev": "cypress run --env testEnv=dev",
+    "test:cypress:staging": "cypress run --env testEnv=staging",
```
[the remark this answers is not in the document]
Comment (tejal-shah) [reply v3/package.json:67]: Sending the testEnv as an environment param makes the cypress tests use the corresponding `config/cypress.<testEnv>.json` which have the environment specific baseUrls configured:
For example, staging's baseUrl is set to `https://codap3.concord.org/index-staging.html` and for dev it is `https://codap3.concord.org/branch/main/`

**Comment by tejal-shah** [comment]
- I've now renamed files, for some to use hyphens and for others made the anme shorter so we don't need underscores or hyphens
- The legend tests were failing because they were visiting the base urls again and hence needed setting the mouseSensor param in their queryParams too - I've fixed this and they should now all pass.

**Review by kswenson — APPROVED** [review]
👍 LGTM

**Pull request merged**